### PR TITLE
fix(bcs): move resource bcs_instance into services and fix codecheck errors

### LINKED ISSUE #2681: BCS instance UT run failed, need to find the reason
```
run acceptance tests of huaweicloud/resource_huaweicloud_bcs_instance_test.go:
=== RUN   TestAccBCSV2Instance_basic
    resource_huaweicloud_bcs_instance_test.go:26: Step 1/1 error: Error running apply: exit status 1
        
        Error: Error waiting for HuaweiCloud CCE cluster to be Available: Resource not found: [GET https://cce.cn-north-4.myhuaweicloud.com/api/v3/projects/0970dd7a1300f5672ff2c003c60ae115/clusters/394dcc20-a15c-11ed-8d34-0255ac100240], error message: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","code":404,"errorCode":"CCE.01404001","errorMessage":"Resource not found.","error_code":"CCE_CM.0003","error_msg":"Resource not found","message":"Cluster not found","reason":"NotFound"}
        
          with huaweicloud_cce_node.test,
          on terraform_plugin_test.tf line 41, in resource "huaweicloud_cce_node" "test":
          41: resource "huaweicloud_cce_node" "test" {
        
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: Error deleting HuaweiCloud CCE Cluster: Resource not found: [DELETE https://cce.cn-north-4.myhuaweicloud.com/api/v3/projects/0970dd7a1300f5672ff2c003c60ae115/clusters/394dcc20-a15c-11ed-8d34-0255ac100240?delete_sfs=true], error message: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","code":404,"errorCode":"CCE.01404001","errorMessage":"Resource not found.","error_code":"CCE_CM.0003","error_msg":"Resource not found","message":"Cluster not found","reason":"NotFound"}
        
--- FAIL: TestAccBCSV2Instance_basic (466.16s)
=== RUN   TestAccBCSV2Instance_kafka
--- PASS: TestAccBCSV2Instance_kafka (2886.12s)
FAIL
FAIL	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud	3352.307s
FAIL
```

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**: move resource bcs_instance into services and fix c odecheck errors

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/bcs" TESTARGS="-run TestAccBCSV2Instance_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/bcs -v -run TestAccBCSV2Instance_basic -timeout 360m -parallel 4
=== RUN   TestAccBCSV2Instance_basic
=== PAUSE TestAccBCSV2Instance_basic
=== CONT  TestAccBCSV2Instance_basic
--- PASS: TestAccBCSV2Instance_basic (1205.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bcs       1205.529s
```

### CODE REVIEW

File: huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go [hunk 1]
```diff
@@ -405,49 +412,50 @@ func resourceBCSInstanceV2Create(d *schema.ResourceData, meta interface{}) error
...
+		d.Set("rollback_support", instance.Basic.IsSupportRollback),
+		d.Set("old_service_version", instance.Basic.OldServiceVersion),
+	)
 
 	portalAddrs := make([]string, len(instance.Basic.AgentPortalAddress))
-	for i, v := range instance.Basic.AgentPortalAddress {
-		portalAddrs[i] = v
-	}
+	copy(portalAddrs, instance.Basic.AgentPortalAddress)
 	d.Set("agent_portal_address", portalAddrs)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go]: all d.Set should use multierror.Append

File: huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go [hunk 2]
```diff
@@ -14,19 +18,21 @@ import (
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/dms"
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
 )
 
-func resourceBCSInstanceV2() *schema.Resource {
+func ResourceInstanceV2() *schema.Resource {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go]: remove V2 in all func names

File: huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go [hunk 3]
```diff
@@ -405,49 +412,50 @@ func resourceBCSInstanceV2Create(d *schema.ResourceData, meta interface{}) error
...
+		d.Set("cross_region_support", instance.Basic.IsCrossRegion),
+		d.Set("rollback_support", instance.Basic.IsSupportRollback),
+		d.Set("old_service_version", instance.Basic.OldServiceVersion),
+	)
 
 	portalAddrs := make([]string, len(instance.Basic.AgentPortalAddress))
-	for i, v := range instance.Basic.AgentPortalAddress {
-		portalAddrs[i] = v
-	}
+	copy(portalAddrs, instance.Basic.AgentPortalAddress)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go]: we can save instance.Basic.AgentPortalAddress directly
array of simple types, like []string and []int can be used in d.Set directly

File: huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go (hunk 1 above)
Comment (wangyuancheng01) [reply huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go]: Has been used multierror.Append

File: huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go (hunk 2 above)
Comment (wangyuancheng01) [reply huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go]: ok, all func names have been removed V2

File: huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go (hunk 3 above)
Comment (wangyuancheng01) [reply huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go]: ok, but lint suggest should use copy(to, from) instead of a loop

Comment (wangyuancheng01) [reply huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go]: done

File: huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go [hunk 4]
```diff
@@ -14,19 +18,21 @@ import (
...
-		Create: resourceBCSInstanceV2Create,
-		Read:   resourceBCSInstanceV2Read,
-		Update: resourceBCSInstanceV2Update,
-		Delete: resourceBCSInstanceV2Delete,
+		CreateContext: resourceBCSInstanceCreate,
+		ReadContext:   resourceBCSInstanceRead,
+		UpdateContext: resourceBCSInstanceUpdate,
+		DeleteContext: resourceBCSInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go:34]: add a description of import in the doc

Comment (wangyuancheng01) [reply huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go:34]: done

**Pull request merged**